docs/internals/controls-package.md

```diff
@@ -9,20 +9,284 @@ description: "The Control Package implements feedback control on an endpoint."
 Each block represents a transfer function of a component.
 In this representation, the control loop is broken down into successive "blocks".
 
-## Example
+## Creating and Using a PID Control Loop
+
+A PID control loop is a commonly used method of controls.
+A PID control loop computes a correction for the error value between SP and PV using three terms:
+
+- A _proportional_ term that is the current error
+- An _integral_ term that is the total cumulative error
+- A _derivative_ term that is the rate of change of the error
+
+By tuning the coefficients on each of these terms, you can adjust how your base converges towards the target value, how quickly the system reaches the target value, and how much the system overshoots when approaching the target value.
+
+The following functions are available for creating and using a control loop:
+
+<!-- prettier-ignore -->
+Method Name | Description
+----------- | -----------
+[`SetupPIDControlConfig`](/internals/controls-package/#setuppidcontrolconfig) | Creates a [PIDLoop](/internals/controls-package/#pidloop) object that contains all the necessary attributes to run a control loop based on the specified [Options](/internals/controls-package/#options).
+[`TunePIDLoop`](/internals/controls-package/#tunepidloop) | Automatically tunes the system and logs the calculated PID values for the loop.
+[`StartControlLoop`](/internals/controls-package/#startcontrolloop) | Starts the control loop in a background thread.
+[`CreateConstantBlock`](/internals/controls-package/#createconstantblock) | Creates a control block of type `constant`, all control loops need at least one constant block representing the set point.
+[`UpdateConstantBlock`](/internals/controls-package/#updateconstantblock) | Updates the value of a constant block to the new set point.
+[`CreateTrapzBlock`](/internals/controls-package/#createtrapzblock) | Creates a control block of type `trapezoidalVelocityProfile`, control loops, such as for motors, that control position need a trapezoidal velocity profile block.
+[`UpdateTrapzBlock`](/internals/controls-package/#updatetrapzblock) | Updates the attributes of a trapezoidal velocity profile block to the new desired max velocity.
+
+### SetupPIDControlConfig
+
+Creates a [PIDLoop](/internals/controls-package/#pidloop) object, which contains all the attributes related to a control loop that a controlled component needs, including, most importantly, the control config.
+
+**Parameters:**
+
+- `pidVals` [([]PIDConfig)](/internals/controls-package/#pidconfig): The P, I, and D values for the control loop, if all are zero the loop will auto-tune and log the calculated PID values.
+- `componentName` [(string)](https://pkg.go.dev/builtin#string): The name of the component that the PID loop controls.
+- `options` [(Options)](/internals/controls-package/#options): All the desired optional parameters to customize the control loop.
```

Comment:
```suggestion
- `options` [(Options)](/internals/controls-package/#options): All the desired [optional parameters](#options) to customize the control loop.
```

Reply:
is it necessary to link twice in the same statement? if this is the norm that's fine I don't feel strongly about it but I'm just wondering if we need the type "Options" and the phrase "optional parameters" to link to the same spot